backend/routes/history.router.js

```diff
@@ -0,0 +1,75 @@
+const express=require("express");
+const { v4: uuidv4 } = require('uuid');
+const router=express.Router();
+const { extend } = require("lodash");
+const {historymodel}=require("../models/history.model.js")
+
+
+
+router.route('/')
+ .get(async (req, res) => {
+   try{
+     const historydata=await historymodel.findOne({historyNo:1});
+     //console.log(history)
+     res.status(200).json({message:"success",historydata:historydata.history})
+   }
+   catch (error){
+     res.status(500).json({success:500,message:"unable to get history",errormessage:error.message})
+   }
+  
+})
+.post(async(req, res) => {
+  try{
+  let {historyobj}=req.body;
+  console.log("historyobj",historyobj)
+  let previtems=await historymodel.findOne({historyNo:1}) ; 
+  if(!previtems){
+    await historymodel.create({historyNo:1,history:[]});
+    let previtems=await historymodel.findOne({historyNo:1}) ; 
```

Comment:
Since you've already run `findOne` once in `line 25`, there's no need to find it again. 
You can update the `histoyrobj` while creation on new `historymodel` in line 27.